frontend/src/pages/Graphing/components/Graph.tsx

```diff
@@ -165,15 +210,121 @@ export const CustomXAxisTick = ({
 export const isActive = (spotlight: number | undefined, idx: number) =>
 	spotlight === undefined || spotlight === idx
 
+export const getViewConfig = (
+	viewType: string,
+	display: string | undefined,
+	nullHandling: string | undefined,
+): ViewConfig => {
+	let viewConfig: ViewConfig
+	if (viewType === 'Line chart') {
+		viewConfig = {
+			type: viewType,
+			showLegend: true,
+			display: display as LineDisplay,
+			nullHandling: nullHandling as LineNullHandling,
+		}
+	} else if (viewType === 'Bar chart') {
+		viewConfig = {
+			type: viewType,
+			showLegend: true,
+			display: display as BarDisplay,
+		}
+	} else if (viewType === 'Table') {
+		viewConfig = {
+			type: viewType,
+			showLegend: false,
+			nullHandling: nullHandling as TableNullHandling,
+		}
+	} else {
+		viewConfig = {
+			type: 'Line chart',
+			showLegend: true,
+		}
+	}
+	return viewConfig
+}
+
 const Graph = ({
-	data,
-	xAxisMetric,
-	yAxisMetric,
-	yAxisFunction,
+	productType,
+	projectId,
+	startDate,
+	endDate,
+	query,
+	metric,
+	functionType,
+	groupByKey,
+	bucketByKey,
+	bucketCount,
+	limit,
+	limitFunctionType,
+	limitMetric,
 	title,
-	loading,
 	viewConfig,
+	onShare,
+	onExpand,
+	onEdit,
 }: ChartProps<ViewConfig>) => {
+	const [graphHover, setGraphHover] = useState(false)
+	const queriedBucketCount = bucketByKey !== undefined ? bucketCount : 1
```

Comment:
Not sure if this is relevant, but if `bucketByKey` is `null`, then `null !==  undefined` will return `true`

However, we can do `!=` to capture null and undefined values. This will still let `""` and `0` evaluate to true

Reply:
I think in this case bucketByKey is `string | undefined` so the original check should work